tests/test_eic.cc

```diff
@@ -0,0 +1,269 @@
+  EcoolRateParas ecool_rate_paras(n_tr, n_long);
+
+  double rate_x, rate_y, rate_s;
+            
+  ForceParas force_paras(ff);
+  force_paras.set_do_test(true);
+  ecooling_rate(ecool_rate_paras, force_paras, ion_beam, cooler, e_beam, ring, rate_x, rate_y, rate_s);
+
+  return;
+}
+
+//A method to sort rows of a matrix based on a specific column.
+// This modifies the original matrix
+void sortrows(std::vector<std::vector<double>>& matrix, int col) {    
+    std::sort(matrix.begin(),
+              matrix.end(),
+              [col](const std::vector<double>& lhs, const std::vector<double>& rhs) {
+                  return lhs[col] > rhs[col];
+              });
+}
+
+//A standard C++ CSV parser to handle the input files we'll need
+// for these comparisons
+vector< vector<double> > ReadCSV(string filename){
+    
+    std::ifstream infile(filename);
+
+    string line;
+    vector<double> vec;
+    vector< vector<double> > matrix;
+
+    while( std::getline(infile,line) ){
+        std::stringstream ss(line);
+        vector<double> row;
+        std::string data;
+        while (getline(ss,data,',') ){
+
+            //Handle lines that don't match the format
+            try{
+                row.push_back( stod(data) );
+            }
+            catch(const std::exception& e){
+             //It was a bad line, we'll toss the whole row   
+                continue;
+            }
+        }
+        //The variables are F_const, v_trans, v_long, electron density, force_trans, force_long
+        if (row.size()  == 6) matrix.push_back(row);
+    }    
+
+    //now that we've ingested the whole file,
+    // make cuts to get the range we want to test.
+
+    vector< vector<double> > out_matrix;
+
+    for(int i=0;i<matrix.size();i++){
+        vector<double> vec = matrix[i];
+        if(vec.size()>0){
+            if (vec[1] < 1){ //Pick a specific low v_transverse
+                out_matrix.push_back(vec);
+            }
+        }
+    }
+    
+    //Now we need to sort by v_long and return it to the
+    // comparison function
+    sortrows(out_matrix,2);
+    
+    return out_matrix;
+}
+
+//Parse the file generated in SetupModel and compare it to the 
+// 'golden' dataset that is stored in the data directory
+double CompareOutput(string filename_golden,string filename_test){
+
+    vector< vector<double> > golden = ReadCSV(filename_golden);
+    vector< vector<double> > test = ReadCSV(filename_test);
+
+    //Calculate the difference between these trends, so there's just
+    // one trend to fit    
+    vector<double> x_vec;
+    vector<double> y_vec;
+    vector<double> y_vec_g;
+    int n = 0;
+
+    for(int i = 0; i<golden.size(); i++){
+        
+        double v_long_g = golden[i][2];
+        double f_long_g = golden[i][5];
+     
+        //Find the matching point for the test dataset
+        // & add it to the fit vector. Our data is already
+        // sorted by v_long.
+        double min = 10000; //The closest point
```

Comment:
10000 seems magic. Should it be something like DBL_MAX? Then down below there should be a check that min is not DBL_MAX.